lib/features/modeling/behavior/SetCompensationActivityAfterPasteBehavior.js

```diff
@@ -0,0 +1,67 @@
+import inherits from 'inherits-browser';
+
+import CommandInterceptor from 'diagram-js/lib/command/CommandInterceptor';
+
+import { getBusinessObject, is } from '../../../util/ModelUtil';
+
+import { hasEventDefinition } from '../../../util/DiUtil';
+
+/**
+ * @typedef {import('diagram-js/lib/core/EventBus').default} EventBus
+ * @typedef {import('../../rules/BpmnRules').default} BpmnRules
+ * @typedef {import('../Modeling').default} Modeling
+ */
+
+
+/**
+ * A behavior that sets the property of Compensation Activity after paste operation
+ *
+ * @param {EventBus} eventBus
+ * @param {Modeling} modeling
+ */
+export default function SetCompensationActivityAfterPasteBehavior(eventBus, modeling) {
+
+  CommandInterceptor.call(this, eventBus);
+
+  this.postExecuted('elements.create', function(event) {
+    const context = event.context,
+          elements = context.elements;
+
+    // check if compensation activity is connected to compensation boundary event
+    for (const element of elements) {
+      if (isForCompensation(element) && !isConnectedToCompensationBoundaryEvent(element)) {
+        modeling.updateProperties(element, { isForCompensation: undefined });
+      }
+    }
+  });
+}
+
+inherits(SetCompensationActivityAfterPasteBehavior, CommandInterceptor);
+
+SetCompensationActivityAfterPasteBehavior.$inject = [
+  'eventBus',
+  'modeling'
+];
+
+
+// helpers //////////////////////
+
+function isForCompensation(element) {
+  const bo = getBusinessObject(element);
+  return bo && typeof bo.get === 'function' && bo.get('isForCompensation');
```

Comment:
Not sure what we want to guard with `bo.get === 'function'.

If there is anything specific, let's create a test case. Otherwise, let's assume that the BO is always a moddle element (flowElement) and has a `.get` method.

Reply:
The issue was caused by some label elements that were breaking this function. I have updated the PR so that now only Task elements are checked for the compensation property.

Reply:
Cf. https://github.com/bpmn-io/bpmn-js/pull/2220/commits/61d9c02086a3184cd33dd3417eb12e5325c0ea8c. The test case was broken. For bpmn-js, a business object must always exist.

Also, you don't have to rely on `get`, but check the property directly. This is what I changed.